tests/test_0835-datetime-type.py

```diff
@@ -498,12 +498,12 @@ def test_highlevel_min_max_axis_None():
             [
                 np.datetime64("2020-03-27T10:41:11"),
                 np.datetime64("2020-01-27T10:41:11"),
-                np.datetime64("2020-05"),
+                np.datetime64("2020-05-27T10:41:11"),
```

Comment:
This had been a test of mixed datetime units; now all the datetimes in this array have the same units and it's no longer a union array.

Okay: reductions of unions with `axis=None` worked before this PR (not any other `axis`, though!) and this PR is removing that capability, but it's good because we're bringing everything in line with the same set of capabilities.